Automattic/sensei
# Add changelogger

This PR copies the implementation from sensei-pro (https://github.com/Automattic/sensei-pro/pull/1185).

### Changes proposed in this Pull Request

* Use the [Jetpack changelogger](https://github.com/Automattic/jetpack-changelogger) tool for creating the changelog entries. Using the same format as [sensei-pro](https://github.com/Automattic/sensei-pro/blob/trunk/changelog.txt). 
* Update the changelog format in changelog.txt and readme.txt to the new one.
* Update `Sensei_Updates` so it can work with the new changelog format.
* Add CI for ensuring that a changelog entry is created or modified on every PR.

### Testing instructions

* Run `./vendor/bin/changelogger add` to interactively create a changelog entry.
* Run `./vendor/bin/changelogger write` to create a new section in the changelog.txt file with all of the entries. You can use `./vendor/bin/changelogger write --help` to see the various options that are available.
* Create a test PR based on this branch and ensure the [changelog CI fails](https://github.com/Automattic/sensei/actions/runs/4078405647/jobs/7028576870) if there is no changelog file added.
* Add a changelog file and make sure the changelog CI job is green.
* Copy the content of readme.txt to [wpreadme.com](https://wpreadme.com/) and make sure the changelog looks good.
